source/speech/__init__.py

```diff
@@ -211,19 +211,17 @@ def speakSpelling(
 	speak(seq, priority=priority)
 
 
-def getSpellingSpeechAddCharMode(
+def _getSpellingSpeechAddCharMode(
 		seq: Generator[SequenceItemT, None, None],
 ) -> Generator[SequenceItemT, None, None]:
 	"""Inserts CharacterMode commands in a speech sequence generator to ensure any single character
 	is spelt by the synthesizer.
 	@param seq: The speech sequence to be spelt.
 	"""
-	synth = getSynth()
-	synthConfig = config.conf["speech"][synth.name]
 	charMode = False
 	for item in seq:
 		if isinstance(item, str):
-			if len(item) == 1 and synthConfig["useSpellingFunctionality"]:

```

Comment:
I have looked at it again and I did not find anything letting me think that it would change the behaviour.
If you make a comparison between the initial code (master branch) and this branch, you will have to consider the following key point:
The variables `charMode` and `synthConfig["useSpellingFunctionality"]` are both used only to add `CharacterModeCommand` commands in the speech sequence. The addition of `CharacterModeCommand` commands has just been extracted in a separate function in the new code.

Please let me know if I missed something or if something more specific is unclear to you.